src/main/java/com/redislabs/redisai/RedisAI.java

```diff
@@ -1,355 +1,424 @@
+                    .getStatusCodeReply().equals("OK");
+
+        } catch (JedisDataException ex) {
+            throw new RedisAIException(ex);
+        }
+    }
+
+    /**
+     * Direct mapping to AI.SCRIPTSET
+     *
+     * @param key    name of key to store the Script in RedisAI server
+     * @param script the Script Object
+     * @return true if Script was properly set in RedisAI server
+     */
+    public boolean setScript(String key, Script script) {
+        try (Jedis conn = getConnection()) {
+            ArrayList<byte[]> args = script.getScriptSetCommandBytes(key);
+            return sendCommand(conn, Command.SCRIPT_SET, args.toArray(new byte[args.size()][]))
+                    .getStatusCodeReply().equals("OK");
+
+        } catch (JedisDataException ex) {
+            throw new RedisAIException(ex);
+        }
     }
-  }
-  
-  /**
-   * AI.SCRIPTSET script_key device script_source
-   */
-  public boolean setScript(String key, Device device, String script){
-
-    try (Jedis conn = getConnection()) {
-
-      ArrayList<byte[]> args = new ArrayList<>();
-      args.add(SafeEncoder.encode(key));
-      args.add(device.getRaw());
-      
-      args.add(Keyword.SOURCE.getRaw());
-      args.add(SafeEncoder.encode(script));
-      
-      return sendCommand(conn, Command.SCRIPT_SET, args.toArray(new byte[args.size()][]))
-          .getStatusCodeReply().equals("OK");
-      
-    } catch(JedisDataException ex ) {
-      throw new RedisAIException(ex);
+
+    /**
+     * Direct mapping to AI.SCRIPTGET
+     *
+     * @param key name of key to get the Script from RedisAI server
+     * @return Script
+     */
+    public Script getScript(String key) throws JRedisAIRunTimeException {
+        try (Jedis conn = getConnection()) {
+            List<?> reply = sendCommand(conn, Command.SCRIPT_GET, SafeEncoder.encode(key), Keyword.META.getRaw(), Keyword.SOURCE.getRaw()).getObjectMultiBulkReply();
+            if (reply.isEmpty()) {
+                return null;
+            }
+            return Script.createScriptFromRespReply(reply);
+        }
+    }
+
+    /**
+     * Direct mapping to AI.SCRIPTDEL
+     *
+     * @param key name of key to delete the Script
+     * @return true if Script was properly delete in RedisAI server
+     */
+    public boolean delScript(String key) {
+
+        try (Jedis conn = getConnection()) {
+            return sendCommand(conn, Command.SCRIPT_DEL, SafeEncoder.encode(key))
+                    .getStatusCodeReply().equals("OK");
+        } catch (JedisDataException ex) {
+            throw new RedisAIException(ex);
+        }
+    }
+
+    /**
+     * AI.MODELRUN model_key INPUTS input_key1 ... OUTPUTS output_key1 ...
+     */
+    public boolean runModel(String key, String[] inputs, String[] outputs) {
+
+        try (Jedis conn = getConnection()) {
+
+            ArrayList<byte[]> args = new ArrayList<>();
```

Comment:
If the method in `List`  is satisfied your needs. I recommend use `List<byte[]> args = new ArrayList<>();` . It's more like programming to an interface. The same advice is offered elsewhere. WDYT?

Reply:
solved in the latest commit